misc/windows-enospc/passthrough.py

```diff
@@ -0,0 +1,36 @@
+"""
+Writing to non-blocking pipe can result in ENOSPC when using Unix APIs on
+Windows.  So, this program passes through data from stdin to stdout, using
+Windows APIs instead of Unix-y APIs.
+"""
+
+from twisted.internet.stdio import StandardIO
+from twisted.internet import reactor
+from twisted.internet.protocol import Protocol
+from twisted.internet.interfaces import IHalfCloseableProtocol
+from twisted.internet.error import ReactorNotRunning
+from zope.interface import implementer
+
+@implementer(IHalfCloseableProtocol)
+class Passthrough(Protocol):
+    def readConnectionLost(self):
+        self.transport.loseConnection()
+
+    def writeConnectionLost(self):
+        try:
+            reactor.stop()
+        except ReactorNotRunning:
+            pass
+
+    def dataReceived(self, data):
+        self.transport.write(data)
+
+    def connectionLost(self, reason):
+        try:
+            reactor.stop()
+        except ReactorNotRunning:
+            pass
+
+
+std = StandardIO(Passthrough())
```

Comment:
So ... could trial use `StandardIO` to manage its interaction with stdout and solve this problem itself?

Not an objection to this change, just a question.

Reply:
Haha - I bet not, because even if it's fixed in trial, trial is just writing to tox and then tox is writing to its own stdout using broken Python APIs.

Reply:
Tox might not do any reading or writing, it might just share the same Windows-equivalent-of-file-descriptor with the subprocess. In which case Trial could use this as fix, yes.